src/main/java/com/volasoftware/tinder/services/EmailSenderService.java

```diff
@@ -0,0 +1,53 @@
+package com.volasoftware.tinder.services;
+
+import com.volasoftware.tinder.model.User;
+import com.volasoftware.tinder.model.Verification;
+import jakarta.mail.MessagingException;
+import jakarta.mail.internet.AddressException;
+import jakarta.mail.internet.InternetAddress;
+import jakarta.mail.internet.MimeMessage;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+@Service
+public class EmailSenderService {
+
+    private JavaMailSender mailSender;
+    private final ResourceLoader resourceLoader;
+
+    public EmailSenderService(JavaMailSender mailSender, ResourceLoader resourceLoader) {
+        this.mailSender = mailSender;
+        this.resourceLoader = resourceLoader;
+    }
+
+    public void sendEmail(User user,
+                          Verification token,
+                          String sender,
+                          String subject,
+                          String contentType) throws MessagingException, IOException {
+        MimeMessage message = mailSender.createMimeMessage();
+        message.setFrom(new InternetAddress(sender));
+        message.setRecipients(MimeMessage.RecipientType.TO,user.getEmail());
+        message.setSubject(subject);
+        message.setContent(getEmailContent(token.getToken()),contentType);
+        mailSender.send(message);
+    }
+
+    private String getEmailContent(String token) throws IOException{
+
+        Resource emailResource = resourceLoader.getResource("classpath:emailResources/ConfirmationPage.html");
```

Comment:
Probable is better to not depend on a resource here. All logic about getting an content that will be send to email (or whatever) is user/token domain specific, no any real connection about sending email.

My advice is to move somewhere else the whole logic and to add one parameter more as content (just an example name) to sendEmail() above